src/app/clusters/scenes/ExtensionFieldsSets.h

```diff
@@ -0,0 +1,100 @@
+/**
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        https://urldefense.com/v3/__http://www.apache.org/licenses/LICENSE-2.0__;!!N30Cs7Jr!UgbMbEQ59BIK-1Xslc7QXYm0lQBh92qA3ElecRe1CF_9YhXxbwPOZa6j4plru7B7kCJ7bKQgHxgQrket3-Dnk268sIdA7Qb8$
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/util/config.h>
+#include <lib/core/CHIPError.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/core/TLV.h>
+
+namespace chip {
+namespace scenes {
+
+typedef struct fieldSets_s
```

Comment:
This leaks every scene-capable clusters together, and may cause binary imcompatibility across changes to configuration. Please find a way to have extension field sets per cluster type.

Also, this doesn't seem to account at all for the presence/absence of a field in a scene, and causes each scene to store all extension fields, not just the ones required by the scene (scenes allow setting which clusters have extension fields used).